araproc/analysis/standard_reco.py

```diff
@@ -389,6 +396,76 @@ def lookup_arrival_time(self,
                                                                                        )
         return arrival_time
 
+    def __get_correlation_function(self, ch1, ch2, event_number):
+        """
+        Returns the correlation function for the channel pair ch1-ch2.
+
+        Parameters
+        ----------
+        ch1 : int
+          Number of first channel in correlation pair.
+        ch2 : int
+          Number of second channel in correlation pair.
+        event_number : int
+          Number of event whose correlation function is being requested.
+
+        Returns
+        -------
+        corr_func : TGraph
+          Cross-correlation function for requested channel pair.
+        """
+
+        if(self.__latest_event_num == -1):
+          raise Exception("No event loaded or event number not set. You must \
+                           run standard_reco.do_standard_reco(wave_bundle, event_num) \
+                           before calling this function. Abort.")
```

Comment:
Do you insist on this? I can imagine worlds where someone is interested in the corr functions without doing the reconstruction. Why not break off populating `self.__corr_functions_v` and `self.__corr_functions_h` into its own function, which gets called either by (1) `reco.get_standard_reco()` or `reco.__get_correlation_function()`. Makes things slightly more modular. But I won't insist on this. It certainly adds complexity.

Reply:
Okay ya I will give it a shot. I can see the benefit!

Reply:
Okay I think I did it. Let me know what you think.